itests/contracts/TransientStorageTest.sol

```diff
@@ -0,0 +1,165 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.24;
+
+contract TransientStorageTest {
+    constructor() {
+    }
+
+    // Test 0: Initial Constructor Test
+    function runTests() public returns (bool) {
+        _runTests();
+    }
+
+    function _runTests() internal {
+        testBasicFunctionality();
+        testLifecycleValidation();
+    }
+
+    // Test 1: Basic Functionality
+    function testBasicFunctionality() public {
+        uint256 slot = 1;
+        uint256 value = 42;
+
+        // Store value using TSTORE
+        assembly {
+            tstore(slot, value)
+        }
+
+        // Retrieve value using TLOAD
+        uint256 retrievedValue;
+        assembly {
+            retrievedValue := tload(slot)
+        }
+
+        require(retrievedValue == value, "TLOAD did not retrieve the correct value");
+
+        // Verify TLOAD from uninitialized location
+        uint256 uninitializedSlot = 2;
+        uint256 uninitializedValue;
+        assembly {
+            uninitializedValue := tload(uninitializedSlot)
+        }
+
+        require(uninitializedValue == 0, "Uninitialized TLOAD did not return zero");
```

Comment:
this is normal behaviour I assume, there's no error condition or way of checking other than a zero value?

Reply:
Yup - uninitialized data defaults to 0 - there's no real way to determine if it's uninitialized or set to 0 intentionally afaik